hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -188,6 +330,48 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
         tableIdentifier.name());
   }
 
+  public  Map<String, String> convertToMetadata(Database database) {
+    Map<String, String> nameMap = Maps.newHashMap();
+
+    nameMap.put("name", database.getName());
+    nameMap.put("location", database.getLocationUri());
+    nameMap.put("comment", database.getDescription());
+    database.getParameters().forEach(nameMap::put);
+
+    return nameMap;
+  }
+
+  public  Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
+    String warehouseLocation = conf.get("hive.metastore.warehouse.dir");
+
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+    if (!meta.containsKey("location")) {
+      Preconditions.checkNotNull(warehouseLocation,
+          "Warehouse location is not set: hive.metastore.warehouse.dir=null");
+    }
+
+    Database database  = new Database();
+    Map<String, String> parameter = Maps.newHashMap();
+
+    database.setName(namespace.toString());
+    database.setLocationUri(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())) + ".db");
```

Comment:
Since we know that `namespace.levels()` has only one entry, this should be `namespace.level(0) + ".db"`  instead of using `SLASH.join`.